core/src/main/java/org/jruby/RubyEnumerator.java

```diff
@@ -473,8 +473,8 @@ private SizeFn enumSizeFn(final ThreadContext context) {
         final RubyEnumerator self = this;
         return new SizeFn() {
             @Override
-            public IRubyObject size(IRubyObject[] args) {
-                return self.size(context);
+            public IRubyObject size(ThreadContext context, IRubyObject self, IRubyObject[] args) {
+                return this.size(context, self, args);
```

Comment:
will recurse infinitely and thus `fail` ... I see why you had `self1` now - makes sense to have a different name here
if I am reading this right? could have used a different name e.g. `recv` if it conflicts (doesn't matter much though) ...